app/jobs/performance_report_job.rb

```diff
@@ -0,0 +1,18 @@
+class PerformanceReportJob < ApplicationJob
+
+    queue_as :performance_report
+  
+    def perform(report_service_string, report_job_guid, period_start_ts, period_end_ts, offset)
+        RavenContextProvider.set_context
+        begin
+            report_service_class = report_service_string.constantize
```

Comment:
I was a bit confused here as we are getting a class from a `String`. I think the word string is misleading in a way as it is the datatype rather than the attribute of the argument passed in. A clearer argument name might be `report_service_name` or `report_service_class_name`. 

This way it is clearer when you call `.constantize` that you want to get the class name and then instantiate it.

Maybe:

```ruby
report_service_class_name = report_service_class_name.constantize
report_service = report_service_class_name.new()
# etc...
```